### ISSUE TITLE
After using Transformer on group further Transformers on its children are wrongly applied 

### ISSUE THREAD

**Issue by Ivanca** [konvajs/konva#602]
![wutwut](https://user-images.githubusercontent.com/907138/54094526-97e04900-436f-11e9-9d38-6f8f24019b7f.gif)

**Comment by Ivanca** [comment]
@lavrton  
<img src="https://media.tenor.com/images/7aa7393ccb90ca13309b41591d8deb3b/tenor.gif" />

**Comment by lavrton** [comment]
The workaround is to use the SAME parent for `Konva.Transformer` and transformed node.

https://jsbin.com/yejavatare/1/edit?html,js,output

But yeh, we should fix the bug.

**Comment by VividWombat** [comment]
I'm using the react demo for transformer, and I get the same result. transformer in the top corner. my nodes are group/image or group/rect&text. so should I add transformer to group node instead of the layer ? currently the group nodes and transformer are all in the same layer

**Comment by VividWombat** [comment]
https://codesandbox.io/s/zx2nl71ll4

this is react transformer demo, with just a group/rect/text added

**Comment by zhuyan0302** [comment]
@Ivanca can you let me see you code ？I have the same problem

**Comment by Ivanca** [comment]
> @Ivanca can you let me see you code ？I have the same problem

Can't share it.

> The workaround is to use the SAME parent for `Konva.Transformer` and transformed node.
> 
> https://jsbin.com/yejavatare/1/edit?html,js,output
> 
> But yeh, we should fix the bug.

Any time estimate for the bug fix, one month? 3 months? less than a year? totally unknown?

**Comment by lavrton** [comment]
Supporting different parents is harder than I expected. So for now using the same parent is the only supported way. I will add a warning into `Konva.Transformer` when it detects different parents.

**Comment by Ivanca** [comment]
@lavrton  If possible could you please post a jsbin example of how to create a temporal parent for Konva.Transformer and the element being transformed (while the final user is doing the transformation) and back? I tried but the scalings and positions get all messed up.

**Comment by lavrton** [comment]
I don't see why you may need to create a new parent. I think this will work just fine:

```javascript
const tr = new Konva.Transformer({
  node: shape
});
shape.getParent().add(tr);
```

**Comment by VividWombat** [comment]
using same parent solved the issue I had

**Comment by rubenski050** [comment]
@VividWombat could you please update https://codesandbox.io/s/zx2nl71ll4 how did you solve the problem. I'm facing the same issue where the transformer is in the corner, apart from the actual rectangle

**Comment by rubenski050** [comment]
nvm. found it. thanks for the fix. 
https://codesandbox.io/s/jlvm840583

**Comment by lavrton** [comment]
Just released `konva@5.0.0`. The bug is fixed.

https://codesandbox.io/s/react-konva-transformer-nx4vm?file=/src/index.js
